packages/format-message/react.js

```diff
@@ -3,11 +3,18 @@
 var React = require('react')
 var formatChildren = require('./base-format-children')
 
+var __counter__ = 0
+
+function makeKey (element) {
+  return element.type + '::' + (__counter__++).toString(16)
+}
```

Comment:
I realize that there are a lot of numbers to go through before you could run into MAX_VALUE, but `Number.MAX_VALUE + 1 === Number.MAX_VALUE`. There is no wraparound in js, so if a page rendered a lot, and was left open long enough, you could run into really weird behavior, since all the keys would start matching.

In addition you are going to unnecessarily replace dom nodes since every render will use a new key, even when the node is exactly the same.